### PR TITLE
Remove code injection. Use message passing instead.

### PR DESCRIPTION
**Pull request by rchrd2**

### CODE REVIEW

**Comment by wagnerand** [comment]
You want to make sure the url is actually a valid url, and not some `javascript:...` which would allow remote code injection with XSS again.

File: src/scripts/client.js [hunk 1]
```diff
@@ -171,26 +175,28 @@
...
-    if (wayback_page_url && wayback_response) {
-      // Some pages use javascript to update the dom so poll to ensure
-      // the banner gets recreated if it is deleted.
-      enforceBannerInterval = setInterval(function() {
-        createBanner(wayback_response.archived_snapshots.closest.url);
-      }, 500);
+  function checkIt(wayback_url) {
+    // Some pages use javascript to update the dom so poll to ensure
+    // the banner gets recreated if it is deleted.
+    enforceBannerInterval = setInterval(function() {
```
Comment (wagnerand) [inline src/scripts/client.js]: I know that's not new code, but you want to cancel the timer in `onunload`.

Comment (rchrd2) [inline src/scripts/client.js]: I can add that, but what's the benefit?

Comment (wagnerand) [inline src/scripts/client.js]: It's generally considered good practice to clean up after yourself. It also prevents memory leaks, although I think if the page object in memory dies, the timer would die as well.

**Comment by rchrd2** [comment]
@wagnerand I updated it to have URL validation like you suggested. I moved that code into it's own function. Let me know if you notice anything else. Thanks.

File: src/scripts/background.js [hunk 2]
```diff
@@ -44,21 +46,44 @@ function wmAvailabilityCheck(url, onsuccess, onfail) {
...
+}
+
+/**
+ * Makes sure response is a valid URL to prevent code injection
+ * @param url {string}
+ * @return {bool}
+ */
+function isValidSnapshotUrl(url) {
+  return ((typeof url) === "string" &&
+    (url.indexOf("http://") === 0 || url.indexOf("https://") === 0));
```
Comment (wagnerand) [inline src/scripts/background.js]: Nitpick: This could be `/^https?:\/\//.test(url)`.

**Comment by wagnerand** [comment]
Looks great, thank you!

**Pull request merged**